### ISSUE TITLE
Scroll down mode stuck. v.1.49

### ISSUE THREAD

**Issue by kuzzmi** [philc/vimium#1549]
When I scroll down, time by time the scrolling mode is kind of stuck, so I can release a button and the scrolling will continue until I press `k`.
```
Vimium: 1.49
Browser: Google Chrome; 41.0.2272.76 (Official Build) 
```

**Comment by smblott-github** [comment]
Thanks for the report, @kuzzmi.  I'm not sure what this could be.  Let's leave this issue sitting and see if we get more reports.  A reliable way to reproduce the problem would be helpful.

**Comment by kuzzmi** [comment]
Unfortunately, I've tried to find a way to reproduce, but it happens occasionally, but more often when I click the key 2 times in a row very quickly. I'll try to look, if that happens without "smooth scrolling"

**Comment by biswanaths** [comment]
Any particular site you faced this problem ?

**Comment by smblott-github** [comment]
@kuzzmi.  Is your machine particularly old?  Or the website you're using particularly busy (slow to load)?  It could be you're uncovering a race condition which only arises under heavy load (although I can't see one in the code).

**Comment by kuzzmi** [comment]
@smblott-github , no, ma machine is new and shiny, the bug is REALLY floating, I can't detect the reason and the way how to reproduce. I've created a new user for Chromium, made a long plain HTML page >10k px height and it worked correctly 99% times (100 key presses, 99 ok, 1 failed). I've created 2 virtual machines with Windows and MacOS and tested there, Windows version worked perfectly, MacOS - the same 99/100. :confused:

**Comment by smblott-github** [comment]
If you have the patience, @kuzzmi, would you be able to try [this version](https://drive.google.com/open?id=0B-PWDBroAYxrOEdCSjdwWEJzd1k&authuser=0), please?

I've stared at the code and can't see where this can happen.  The version above logs the variables which affect scrolling to the console (9898473ac0f7ba482f920cd0479254fe2256b4f1), and is based off master.

(Disable your regular Vimium, and drag and drop that file onto the extensions page.  You might need to enable developer mode, not sure.)

**Comment by kuzzmi** [comment]
​I'll do this for sure, but if you don't mind, I'd like to postpone that
stuff a little bit. I suppose until tomorrow.

Best regards,
Igor Kuzmenko

**Comment by kuzzmi** [comment]
I've found the reason of that behavior. That was enormously stupid... Basically, the first thing I do on a machine is customize the keyrate to a very high repeat speed and very low delay before repeating. So, I've tried that version with debug info and voila! Basically if I press a key once - everything works good, if I release a key and the key repeat happens - occasionally the browser doesn't understand that the key has been released. I think that's not plugin's problem, but the problem with Chromium itself.

**Comment by smblott-github** [comment]
Thanks for the update, @kuzzmi. So, with your keyboard setup, we get a `keydown` event with no corresponding `keyup`, yes?

(How do you configure your keyboard?  Something like `xset r RATE DELAY`?  If so, what params are you using?)

**Comment by kuzzmi** [comment]
Exactly, no `keyup` event.

I use `xset r rate 150 150`

**Comment by smblott-github** [comment]
Thanks for the report and follow up, @kuzzmi.  This looks like a Chromium/XOrg bug, or bad feature interaction, or something. I added a note to the code (https://github.com/philc/vimium/commit/3e1295516025d7ca29dea9fadc9ba75aa91f4bce), just so we know.

(As a workaround -- since you like super-fast keyboard repeat -- you could disable smooth scrolling on the options page and set `ScrollSstepSize` to a small value.  You'd probably get pretty much the same effect as smooth scrolling.)

**Comment by kuzzmi** [comment]
Thanks, @smblott-github. Yep, I'll report the bug for those guys. 
BTW, I've already applied this workaround, for now it works fine.

**Comment by spillevink** [comment]
I'm experiencing this aswell. Would it be possible to stop scroll when s/k is pressed again? Right now the only way to stop the scrolling is to reload the page. If i for example open the omnibar the scrolling will stop, but will continue being stuck if i press j or k.

**Comment by smblott-github** [comment]
@spillevink; scrolling should stop on *any* `keyup`/`keydown` event.  That's not what you're seeing, right?

**Comment by philc** [comment]
I can also confirm this is happening. I will sometimes lift j and the page
keeps scrolling down. Typing another key stops it.

**Comment by spillevink** [comment]
@smblott-github: 99% of the time j/k will not get stuck. But if they do get stuck, they will not unstick even if i press another key. The page will just momentarily stop scrolling if i press another key, but as soon as i try to scroll another 30 pixels by tapping j/k, the scroll will be stuck again as if i was holding the key. The only way I have found to unstuck them is by reloading the page.

**Comment by smblott-github** [comment]
Closing.

This should be fixed in #1930.
